waltz-client/src/main/java/com/wepay/waltz/client/internal/Partition.java

```diff
@@ -269,17 +271,22 @@ public void partitionAhead() {
     public void ensureMounted() {
         if (!mounted) {
             synchronized (lock) {
+                final long due = System.currentTimeMillis() + ENSURE_MOUNTED_TIMEOUT;
                 while (state != PartitionState.CLOSED && !mounted) {
                     if (transactionMonitor.isStopped()) {
                         throw new PartitionInactiveException(partitionId);
                     } else if (clientHighWaterMarkAhead) {
                         throw new IllegalStateException(String.format("client is ahead of store for partition: %d", partitionId));
                     }
-
-                    try {
-                        lock.wait();
-                    } catch (InterruptedException ex) {
-                        Thread.interrupted();
+                    long remaining = due - System.currentTimeMillis();
+                    if (remaining > 0) {
+                        try {
+                            lock.wait(ENSURE_MOUNTED_TIMEOUT);
```

Comment:
Here, do we want to wait on for the remaining amount of time left ?

Reply:
hmm, I am curious, there is no unit test for this public method ?

Reply:
No, the method doesn't have its own unit test. The method is part of a test flow of other tests such these in InternalStreamClientTest.